server/apps/immich/src/api-v1/user/user-repository.ts

```diff
@@ -60,28 +63,12 @@ export class UserRepository implements IUserRepository {
   }
 
   public async create(user: Partial<UserEntity>): Promise<UserEntity> {
-    const localAdmin = await this.getAdmin();
-    if (!localAdmin && !user.isAdmin) {
-      throw new BadRequestException('The first registered account must the administrator.');
-    }
-
-    if (user.password) {
-      user.salt = await bcrypt.genSalt();
-      user.password = await this.hashPassword(user.password, user.salt);
-    }
-
     return this.userRepository.save(user);
   }
 
   public async update(id: string, user: Partial<UserEntity>): Promise<UserEntity> {
     user.id = id;
 
-    // If payload includes password - Create new password for user
-    if (user.password) {
-      user.salt = await bcrypt.genSalt();
-      user.password = await this.hashPassword(user.password, user.salt);
-    }
-
     // TODO: can this happen? If so we can move it to the service, otherwise remove it (also from DTO)
     if (user.isAdmin) {
```

Comment:
This is also business logic that should be in the user core. Also, this throws a BadRequestException, but the repository.updateUser call is wrapped in a try / catch so this will be suppressed and turned into an InternalServerError :disappointed:.

Reply:
Yes you are right (although this was already like this, not introduced in this PR). I'll move it to the user.core including the TODO (not sure if this can happen since there is no test implemented for this case)